### PR TITLE
feat: new subscription category for awaiting deletion events

### LINKED ISSUE #769: A new subscription category for AWAITING_DELETION events
**Acceptance criteria:**
- [ ] create a new `CLEAN_UP` subscription category on the EL;
- [ ] the category finder to pop projects having `AWAITING_DELETION` events
  - [ ] the finder to order projects by `AWAITING_DELETION` events date DESC
  - [ ] the finder to change status of all `AWAITING_DELETION` events to `DELETING`
- [ ] EL to send events to the subscriber
   ```json
   {
    "categoryName": "CLEAN_UP",
    "project": {
      "id": 123,
      "path": "namespace/project-name"
    }
   }
   ```
- [ ] EL keeps a note in the `event_delivery` about the latest `AWAITING_DELETION` event from the project
- [ ] the subscription category to have a dispatch recovery strategy that sends a `EVENTS_STATUS_CHANGE` for the same event as noted in the `event_delivery` (point above)
   ```json
   {
    "categoryName": "EVENTS_STATUS_CHANGE",
    "id":           "12324",
    "project": {
      "id": 123,
      "path": "namespace/project-name"
    },
    "newStatus": "AWAITING_DELETION"
   }
   ```
- [ ] update README

### PR DESCRIPTION
**Pull request by joke1196**
/deploy renku=master renku-ui=master renku-core=v1.0.3 #persist

closes #769

### CODE REVIEW

File: event-log/src/main/scala/io/renku/eventlog/subscriptions/cleanup/DispatchRecovery.scala [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        json"""{
+        "categoryName": "EVENTS_STATUS_CHANGE",
+        "project": {
+          "id":   ${event.project.id},
+          "path": ${event.project.path}
+        },
+          "newStatus": $AwaitingDeletion
+        }"""
+      ),
+      errorMessage = s"${SubscriptionCategory.name}: Marking event as $AwaitingDeletion failed"
```
Comment (jachro) [inline event-log/src/main/scala/io/renku/eventlog/subscriptions/cleanup/DispatchRecovery.scala:53]: Maybe we could be more explicit in the message that we mark one or many project events not a single event.

File: event-log/src/main/scala/io/renku/eventlog/subscriptions/eventdelivery/model.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package io.renku.eventlog.subscriptions.eventdelivery
```
Comment (jachro) [inline event-log/src/main/scala/io/renku/eventlog/subscriptions/eventdelivery/model.scala:1]: Do we have the preamble in this file?

File: event-log/src/main/scala/io/renku/eventlog/TypeSerializers.scala [hunk 3]
```diff
@@ -90,6 +91,11 @@ trait TypeSerializers {
   val eventFailureStatusEncoder: Encoder[EventStatus.FailureStatus] =
     eventStatusEncoder.contramap((s: EventStatus.FailureStatus) => s: EventStatus)
 
+  val eventTypeIdDecoder: Decoder[EventTypeId] = varchar.map { case "DELETING" =>
```
Comment (jachro) [inline event-log/src/main/scala/io/renku/eventlog/TypeSerializers.scala:94]: Should we maybe do the case matching on the `DeletingProjectTypeId.value`?

File: event-log/src/test/scala/io/renku/eventlog/subscriptions/cleanup/DispatchRecoverySpec.scala [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      val exception  = exceptions.generateOne
+      val subscriber = subscriberUrls.generateOne
+
+      val eventRequestContent = EventRequestContent.NoPayload(json"""{
+        "categoryName": "EVENTS_STATUS_CHANGE",
+        "project": {
+          "id":   ${event.project.id},
+          "path": ${event.project.path}
+        },
+        "message": ${EventMessage(exception)},
```
Comment (jachro) [inline event-log/src/test/scala/io/renku/eventlog/subscriptions/cleanup/DispatchRecoverySpec.scala:76]: Do we do anything with the `message` we send here in the event?

File: graph-commons/src/main/scala/io/renku/graph/model/events.scala [hunk 5]
```diff
@@ -67,6 +67,9 @@ object events {
     protected[this] override def maybeMax: Option[Instant] = instantNow.plus(24, HOURS).some
   }
 
+  final class ProjectDeliveryId private (val value: String) extends AnyVal with StringTinyType
```
Comment (jachro) [inline graph-commons/src/main/scala/io/renku/graph/model/events.scala:70]: Is the `ProjectDeliveryId` still needed here?

File: event-log/src/main/scala/io/renku/eventlog/subscriptions/eventdelivery/model.scala (hunk 2 above)
Comment (joke1196) [reply event-log/src/main/scala/io/renku/eventlog/subscriptions/eventdelivery/model.scala:1]: No good catch

File: event-log/src/test/scala/io/renku/eventlog/subscriptions/cleanup/DispatchRecoverySpec.scala (hunk 4 above)
Comment (joke1196) [reply event-log/src/test/scala/io/renku/eventlog/subscriptions/cleanup/DispatchRecoverySpec.scala:76]: No you are right this is not used

File: graph-commons/src/main/scala/io/renku/graph/model/events.scala (hunk 5 above)
Comment (joke1196) [reply graph-commons/src/main/scala/io/renku/graph/model/events.scala:70]: No I'll remove it

**Pull request merged**